modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaRenderedEntityFormatter.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase;
+use Drupal\file\Entity\File;
+use Drupal\media\MediaInterface;
+
+/**
+ * Plugin implementation of the 'Featured media as entity' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "oe_featured_media_entity_view",
+ *   label = @Translation("Rendered entity"),
+ *   description = @Translation("Display the entity of the referenced media and the caption."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaRenderedEntityFormatter extends EntityReferenceFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item): array {
+    $params = [];
+    $params['description'] = $item->caption;
+
+    if (!$item->entity instanceof MediaInterface) {
+      return [];
+    }
+
+    switch ($item->entity->bundle()) {
```

Comment:
Let's not use bundles, but rather media source type, check how we do in `oe_theme_preprocess_paragraph__oe_text_feature_media()`, you can reuse most of that logic (and cache info too).

Also tests for this formatter are missing, you can check what we did to test that paragraph above and, most importantly, how we tested a custom formatter here
```
./modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/SocialMediaLinksFormatterTest.php 
```